examples/example_3d_simulate/example_3d_fit.py

```diff
@@ -1,130 +1,116 @@
-    spatial_model_sherpa.xpos.freeze()
-    spatial_model_sherpa.ypos.freeze()
-    spatial_model_sherpa.r0.freeze()
-    spatial_model_sherpa.width.freeze()
+def load_cubes():
+    npred_cube = WcsNDMap.read('npred.fits')
+    exposure_cube = WcsNDMap.read('exposure.fits')
+    i_nan = np.where(np.isnan(exposure_cube.data))
+    exposure_cube.data[i_nan] = 0
+    return dict(counts=npred_cube, exposure=exposure_cube)
 
-    source_model = CombinedModel3D(
-        spatial_model=spatial_model_sherpa,
-        spectral_model=spectral_model_sherpa,
+def get_fit_model():
+    spatial_model = SkyGaussian2D(
+        lon_0='0 deg',
+        lat_0='0 deg',
+        sigma='1 deg',
     )
-
-    # source_model = CombinedModel3DInt(
-    #     spatial_model=spatial_model_sherpa,
-    #     spectral_model=spectral_model_sherpa,
-    #     exposure=exposure,
-    #     coord=coord,
-    #     energies=energies,
-    #     use_psf=use_psf,
-    #     psf=None,
-    # )
-
-    print(source_model)
-    # source_model_cube = source_model.evaluate_cube(ref_cube)
-    # model = source_model  # + background_model
-
-    # source_model2 = CombinedModel3DInt(
-    #     coord=coord,
-    #     energies=energies,
-    #     use_psf=False,
-    #     exposure=cubes['exposure'],
-    #     psf=None,
-    #     spatial_model=spatial_model,
-    #     spectral_model=spectral_model,
-    # )
-
-    model = 1e-9 * exposure * source_model  # 1e-9 flux factor
-
-    fit = Fit(
-        data=counts,
-        model=model,
-        stat=Chi2ConstVar(),
-        method=LevMar(),
-        # estmethod=Covariance(),
+    spectral_model = PowerLaw(
+        index=2,
+        amplitude='1e-10 cm-2 s-1 TeV-1',
+        reference='1 TeV',
+    )
+    return SkyModel(
+        spatial_model=spatial_model,
+        spectral_model=spectral_model,
     )
 
-
-    fit_results = fit.fit()
-    print(fit_results.format())
-
-    print('------------------------------------ end fitting')
-
-    # import IPython; IPython.embed()
-
+def main():
+    log.setLevel('INFO')
+    log.info('Starting ...')
+
+    cubes = load_cubes()
+    log.info('Loaded cubes: {}'.format(cubes))
+
+    model = get_fit_model()
+    log.info('Loaded model: {}'.format(model))
+
+    # NOTE: Without this the fitter set the amplitude to 0
+    # This result in npred = 0 and thus cash = 0 everywhere
+    model.parameters['amplitude'].parmin = 1e-12
+
+    fit = CubeFit(cubes=cubes, model=model.copy())
+    log.info('Created analysis: {}'.format(fit))
+
+    fit.fit()
+    log.info('Starting values\n{}'.format(get_fit_model().parameters))
+    log.info('Best fit values\n{}'.format(fit.model.parameters))
+    log.info('True values\n{}'.format(get_sky_model().parameters))
+    
+
+class CubeFit(object):
+    """Perform 3D likelihood fit
+
+    This is my first go at such a class. It's geared to the SpectrumFit class
+    which does the 1D spectrum fit.
+
+    Parameters
+    ----------
+    cubes : dict
```

Comment:
Why pass some inputs in a dict, but model separately?
Soon we'll pass other things in, e.g. PSF.
I'd suggest to just pass in all objects separately that we need here, at least for now as long as we have no "dataset" container for this.

Reply:
:+1: